### ISSUE TITLE
[YW] Server provision script fails with "The conditional check 'result.rc == 0' failed.

### ISSUE THREAD

**Issue by ajcaldera1** [yugabyte/yugabyte-db#2273]
While running the server provisioning script on YW 1.3.2.1, Ansible failed with the message: "The conditional check 'result.rc == 0' failed. The error was: error while evaluating conditional (result.rc == 0): 'dict object' has no attribute "rc".

I tried updating `server_base_linux.yml` to use the syntax suggested by: https://github.com/geerlingguy/ansible-role-repo-epel/issues/30 to say `until: result is succeeded`, but I now encounter this issue:

```
Executing provision now for instance with IP cud2-005174.csdev.corp...
Failed!
2019-09-11 20:15:42,896 INFO: Found onprem cloud credentials in env.
2019-09-11 20:15:42,896 INFO: Host @@DEFAULT_NODE_NAME@@ already created.
2019-09-11 20:15:42,897 INFO: Running ansible command ["ansible-playbook" "/opt/yugabyte/devops/yb-server-provision.yml" "--vault-password-file" "/opt/yugabyte/yugaware/data/keys/474b04f0-d1c0-42da-b29c-cc3ade77928f/iaasv3-key.vault_password" "--private-key" "/opt/yugabyte/yugaware/data/keys/474b04f0-d1c0-42da-b29c-cc3ade77928f/iaasv3-key.pem" "--user" "yugabyte" "-i" "cud2-005174.csdev.corp," "-c" "ssh" "--extra-vars" "{\"ansible_port\": 22, \"cloud_type\": \"onprem\", \"server_type\": \"cluster-server\", \"ssd_size_gb\": 250, \"mount_points\": \"/mnt/disk1\", \"instance_search_pattern\": \"all\", \"yb_server_ssh_user\": \"yugabyte\", \"num_volumes\": 10, \"instance_name\": \"@@DEFAULT_NODE_NAME@@\", \"instance_type\": null, \"device_names\": [], \"ssh_user\": \"yugabyte\", \"disk_iops\": 1000, \"yb_ansible_host\": \"cud2-005174.csdev.corp\", \"user_name\": \"yugabyte\", \"vars_file\": \"/opt/yugabyte/yugaware/data/keys/474b04f0-d1c0-42da-b29c-cc3ade77928f/iaasv3-key.vault\", \"local_package_path\": \"/opt/third-party\"}"]

PLAY [Validate server type] ****************************************************

TASK [setup] *******************************************************************
Wednesday 11 September 2019  20:15:44 +0000 (0:00:00.016)       0:00:00.016 ***
ok: [localhost]

TASK [assert] ******************************************************************
Wednesday 11 September 2019  20:15:45 +0000 (0:00:00.904)       0:00:00.920 ***
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

PLAY [Install Python on Ubuntu] ************************************************

TASK [Verify and Install Python] ***********************************************
Wednesday 11 September 2019  20:15:45 +0000 (0:00:00.052)       0:00:00.972 ***
skipping: [cud2-005174.csdev.corp]

PLAY [Provision server] ********************************************************

TASK [setup] *******************************************************************
Wednesday 11 September 2019  20:15:45 +0000 (0:00:00.126)       0:00:01.098 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : assert] ****************************************************
Wednesday 11 September 2019  20:15:58 +0000 (0:00:13.114)       0:00:14.213 ***
ok: [cud2-005174.csdev.corp] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [server_base : assert] ****************************************************
Wednesday 11 September 2019  20:15:58 +0000 (0:00:00.053)       0:00:14.267 ***
ok: [cud2-005174.csdev.corp] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [server_base : Check if /etc/init.d/sshguard exists] **********************
Wednesday 11 September 2019  20:15:58 +0000 (0:00:00.058)       0:00:14.325 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Stop and disable sshguard] *********************************
Wednesday 11 September 2019  20:15:59 +0000 (0:00:00.743)       0:00:15.068 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Check if /etc/ssh/sshd_config exists] **********************
Wednesday 11 September 2019  20:15:59 +0000 (0:00:00.046)       0:00:15.114 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Disable reverse DNS lookup on SSH login] *******************
Wednesday 11 September 2019  20:15:59 +0000 (0:00:00.548)       0:00:15.663 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Allow core dumps] ******************************************
Wednesday 11 September 2019  20:16:00 +0000 (0:00:00.712)       0:00:16.375 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Increase limit on number of open files] ********************
Wednesday 11 September 2019  20:16:01 +0000 (0:00:00.730)       0:00:17.105 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Increase limit on max number of processes] *****************
Wednesday 11 September 2019  20:16:01 +0000 (0:00:00.567)       0:00:17.672 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Remove file /etc/security/limits.d/20-nproc.conf if present] ***
Wednesday 11 September 2019  20:16:02 +0000 (0:00:00.543)       0:00:18.216 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Update package cache on RedHat platforms] ******************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.763)       0:00:18.979 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Set Linux packages to Install] *****************************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.045)       0:00:19.025 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Set Dev packages to Install] *******************************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.054)       0:00:19.079 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Set Debian packages to Install] ****************************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.036)       0:00:19.116 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Set RedHat packages to Install] ****************************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.040)       0:00:19.157 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Find all yum repos] ****************************************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.042)       0:00:19.200 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Replace mirror links from https to http] *******************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.044)       0:00:19.244 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Replace mirror links from https to http] *******************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.052)       0:00:19.296 ***
skipping: [cud2-005174.csdev.corp]

TASK [server_base : Install epel-release if RedHat.] ***************************
Wednesday 11 September 2019  20:16:03 +0000 (0:00:00.050)       0:00:19.347 ***
ok: [cud2-005174.csdev.corp]

TASK [server_base : Install linux packages we need on our YugaByte clusters on CentOS] ***
Wednesday 11 September 2019  20:16:19 +0000 (0:00:15.711)       0:00:35.059 ***
FAILED - RETRYING: TASK: server_base : Install linux packages we need on our YugaByte clusters on CentOS (5 retries left).
FAILED - RETRYING: TASK: server_base : Install linux packages we need on our YugaByte clusters on CentOS (4 retries left).
FAILED - RETRYING: TASK: server_base : Install linux packages we need on our YugaByte clusters on CentOS (3 retries left).
FAILED - RETRYING: TASK: server_base : Install linux packages we need on our YugaByte clusters on CentOS (2 retries left).
FAILED - RETRYING: TASK: server_base : Install linux packages we need on our YugaByte clusters on CentOS (1 retries left).
failed: [cud2-005174.csdev.corp] (item=None) => {"attempts": 5, "failed": true, "item": null, "module_stderr": "Shared connection to cud2-005174.csdev.corp closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_tavUPY/ansible_module_yum.py\", line 1119, in <module>\r\n    main()\r\n  File \"/tmp/ansible_tavUPY/ansible_module_yum.py\", line 1101, in main\r\n    pkg = [ p.strip() for p in params['name']]\r\nTypeError: 'NoneType' object is not iterable\r\n", "msg": "MODULE FAILURE"}
        to retry, use: --limit @/opt/yugabyte/devops/yb-server-provision.retry

PLAY RECAP *********************************************************************
cud2-005174.csdev.corp     : ok=12   changed=0    unreachable=0    failed=1
localhost                  : ok=2    changed=0    unreachable=0    failed=0

Wednesday 11 September 2019  20:16:54 +0000 (0:00:34.888)       0:01:09.948 ***
===============================================================================
server_base : Install linux packages we need on our YugaByte clusters on CentOS -- 34.89s
server_base : Install epel-release if RedHat. -------------------------- 15.71s
setup ------------------------------------------------------------------ 13.11s
setup ------------------------------------------------------------------- 0.90s
server_base : Remove file /etc/security/limits.d/20-nproc.conf if present --- 0.76s
server_base : Check if /etc/init.d/sshguard exists ---------------------- 0.74s
server_base : Allow core dumps ------------------------------------------ 0.73s
server_base : Disable reverse DNS lookup on SSH login ------------------- 0.71s
server_base : Increase limit on number of open files -------------------- 0.57s
server_base : Check if /etc/ssh/sshd_config exists ---------------------- 0.55s
server_base : Increase limit on max number of processes ----------------- 0.54s
Verify and Install Python ----------------------------------------------- 0.13s
server_base : assert ---------------------------------------------------- 0.06s
server_base : Set Linux packages to Install ----------------------------- 0.05s
server_base : assert ---------------------------------------------------- 0.05s
server_base : Replace mirror links from https to http ------------------- 0.05s
assert ------------------------------------------------------------------ 0.05s
server_base : Replace mirror links from https to http ------------------- 0.05s
server_base : Stop and disable sshguard --------------------------------- 0.05s
server_base : Update package cache on RedHat platforms ------------------ 0.05s
Traceback (most recent call last):
  File "/opt/yugabyte/devops/python_virtual_env/bin/ybcloud.py", line 4, in <module>
    __import__('pkg_resources').run_script('ybops==0.4', 'ybcloud.py')
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 666, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1462, in run_script
    exec(code, namespace, namespace)
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/ybops-0.4-py2.7.egg/EGG-INFO/scripts/ybcloud.py", line 15, in <module>
    YbCloud().run()
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/ybops-0.4-py2.7.egg/ybops/cloud/ybcloud.py", line 54, in run
    self.options.func(self.options)
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/ybops-0.4-py2.7.egg/ybops/cloud/common/method.py", line 76, in callback_wrapper
    self.callback(args)
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/ybops-0.4-py2.7.egg/ybops/cloud/onprem/method.py", line 67, in callback
    super(OnPremProvisionInstancesMethod, self).callback(args)
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/ybops-0.4-py2.7.egg/ybops/cloud/common/method.py", line 366, in callback
    self.cloud.setup_ansible(args).run("yb-server-provision.yml", self.extra_vars, host_info)
  File "/opt/yugabyte/devops/python_virtual_env/lib/python2.7/site-packages/ybops-0.4-py2.7.egg/ybops/cloud/common/ansible.py", line 126, in run
    process_args, p.returncode, stderr))
ybops.common.exceptions.YBOpsRuntimeError: Runtime error: Playbook with args ['ansible-playbook', '/opt/yugabyte/devops/yb-server-provision.yml', '--vault-password-file', '/opt/yugabyte/yugaware/data/keys/474b04f0-d1c0-42da-b29c-cc3ade77928f/iaasv3-key.vault_password', '--private-key', '/opt/yugabyte/yugaware/data/keys/474b04f0-d1c0-42da-b29c-cc3ade77928f/iaasv3-key.pem', '--user', u'yugabyte', '-i', u'cud2-005174.csdev.corp,', '-c', 'ssh', '--extra-vars', '{"ansible_port": 22, "cloud_type": "onprem", "server_type": "cluster-server", "ssd_size_gb": 250, "mount_points": "/mnt/disk1", "instance_search_pattern": "all", "yb_server_ssh_user": "yugabyte", "num_volumes": 10, "instance_name": "@@DEFAULT_NODE_NAME@@", "instance_type": null, "device_names": [], "ssh_user": "yugabyte", "disk_iops": 1000, "yb_ansible_host": "cud2-005174.csdev.corp", "user_name": "yugabyte", "vars_file": "/opt/yugabyte/yugaware/data/keys/474b04f0-d1c0-42da-b29c-cc3ade77928f/iaasv3-key.vault", "local_package_path": "/opt/third-party"}'] failed with return code 2 and error

```

**Comment by ajcaldera1** [comment]
Bump.

**Comment by bmatican** [comment]
> Shared connection to cud2-005174.csdev.corp closed.
Is access to yum restricted? @ramkumarvs I thought we had the flow where we do not need sudo, and we can skip these package install steps? cc @ajcaldera1

**Comment by ramkumarvs** [comment]
fixed as part of e8e887a69c
